app/src/main/java/com/kickstarter/ui/adapters/projectcampaign/ViewElementAdapter.kt

```diff
@@ -156,18 +180,81 @@ class ViewElementAdapter(val requireActivity: FragmentActivity) : RecyclerView
             }
         }
 
+        (element as? VideoViewElement)?.let { videoElement ->
+            (viewHolder as? VideoElementViewHolder)?.let {
+                viewHolder.bindData(videoElement)
+            }
+        }
+
         (element as? ExternalSourceViewElement)?.let { externalSourceViewElement ->
             (viewHolder as? ExternalViewViewHolder)?.let {
                 viewHolder.bindData(externalSourceViewElement)
             }
         }
     }
 
+    override fun onViewRecycled(holder: RecyclerView.ViewHolder) {
+
+        (holder as? VideoElementViewHolder)?.let { videoElementViewHolder ->
+            videoElementViewHolder.releasePlayer(index = videoElementViewHolder.bindingAdapterPosition)
+        }
+
+        super.onViewRecycled(holder)
+    }
+
     private enum class ElementViewHolderType {
         TEXT,
         IMAGE,
         VIDEO,
         EMBEDDED,
         EXTERNAL_SOURCES
     }
+
+    // for hold all players generated
+    private var playersMap: MutableMap<Int, SimpleExoPlayer?> = mutableMapOf()
+
+    // for hold current player
+    private var currentPlayingVideo: Pair<Int, SimpleExoPlayer?>? = null
+
+    override fun onDetachedFromRecyclerView(recyclerView: RecyclerView) {
+        super.onDetachedFromRecyclerView(recyclerView)
+        releaseAllPlayers()
+    }
+
+    fun releaseAllPlayers() {
```

Comment:
Execep for the overrided method it seems code related to the video player, and does not belong in the adapter, Can we find a way to handle it on the ViewHolder?